java/ql/lib/semmle/code/java/frameworks/apache/Codec.qll

```diff
@@ -0,0 +1,135 @@
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUsAscii;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf16;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf16Be;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf16Le;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.binary;StringUtils;true;newStringUtf8;(byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;DigestUtils;(MessageDigest);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;getDigest;(String,MessageDigest);;Argument[1];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;getMessageDigest;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,ByteBuffer);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,File);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,InputStream);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,Path,OpenOption[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,RandomAccessFile);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;DigestUtils;true;updateDigest;(MessageDigest,byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;HmacUtils;false;updateHmac;(Mac,InputStream);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;HmacUtils;false;updateHmac;(Mac,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;HmacUtils;false;updateHmac;(Mac,byte[]);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;Md5Crypt;true;md5Crypt;(byte[],String,String);;Argument[2];ReturnValue;taint;",
+        "org.apache.commons.codec.digest;Md5Crypt;true;md5Crypt;(byte[],String,String,Random);;Argument[2];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Lang;true;loadFromResource;(String,Languages);;Argument[1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;from;(Set);;Element of Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;merge;(LanguageSet);;Argument[-1];ReturnValue;value;",
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;merge;(LanguageSet);;Argument[0];ReturnValue;taint;",
```

Comment:
I can't see `merge` in the [up-to-date documentation](https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/language/bm/Languages.LanguageSet.html). Is it from a previous version?

Reply:
Good catch, this is actually package private and should not have been generated.